t/05-distribution.t

```diff
@@ -61,6 +61,37 @@ subtest 'script_run' => sub {
     is $cmdcall->{buffer_size}, 6272, 'appropriate buffer size used for long command';
 };
 
+subtest 'set expected serial and autoinst failures' => sub {
+    my $d = distribution->new;
+    # Define the expected failures data
+    my @failures = (
+        {type => 'soft', message => '%s Failure Message 1', pattern => 'Test Pattern1'},
+        {type => 'soft', message => '%s Failure Message 2', pattern => 'Test Pattern2'},
+        {type => 'hard', message => '%s Failure Message 3', pattern => 'Test Pattern3'},
+    );
+    # Subroutine to generate failure data with formatted messages
+    my sub _generate_failures ($type, %details) {
+        return [
+            map {
+                {
+                    message => sprintf($details{message}, $type),
+                    pattern => qr/$details{pattern}/
+                }
+            } @failures
+        ];
+    }
+    # Set and test serial failures
+    my %failure_hash = (
+        message => "$failures[0]->{message}", pattern => "$failures[0]->{pattern}",
+        pattern => "$failures[0]->{pattern}"
```

Comment:
not sure. why you define pattern twice?

Reply:
Ohh, it is my mistake, let me correct it